File: py/search/array/SequenceSearch.py
#!/usr/bin/env python
# -*- coding:gbk -*-

from typing import List

class SequenceSearch:
    def search(self, nums: List[int], target: int) -> int:
        aux = list(nums)
        aux.append(target)

        index = 0
        while target != aux[index]:
            index += 1
        return index if index < len(nums) else -1

File: cpp/search/array/SequenceSearch.cpp
#include <iostream>
#include <vector>
using namespace std;

class SequenceSearch {
public:
    int search(vector<int> &nums, int target) {
        vector<int> aux(nums.begin(), nums.end());
        aux.emplace_back(target);

        int index = 0;
        while (target != aux[index]) {
            ++index;
        }
        return (index < (int) nums.size()) ? index : -1;
    }
};
